diff --git a/dist/chunk-DSMOQWPH.js b/dist/chunk-DSMOQWPH.js
index c9e15ab14c7501c99740f32fb39e2373c8bdaa1f..ad99b3640d3166e9eca3b3520066fff8b5697e4d 100644
--- a/dist/chunk-DSMOQWPH.js
+++ b/dist/chunk-DSMOQWPH.js
@@ -2277,7 +2277,7 @@ function ChainModal({ onClose, open }) {
   }, [activeConnector, onClose, stopSwitching]);
   useEffect12(() => {
     if (networkError && networkError.name === "UserRejectedRequestError") {
-      stopSwitching();
+      // stopSwitching();
     }
   }, [networkError, stopSwitching]);
   const { appName } = useContext9(AppContext);
@@ -2285,7 +2285,7 @@ function ChainModal({ onClose, open }) {
     return null;
   }
   return /* @__PURE__ */ React36.createElement(Dialog, {
-    onClose,
+    onClose: stopSwitching,
     open,
     titleId
   }, /* @__PURE__ */ React36.createElement(DialogContent, {
@@ -2383,7 +2383,16 @@ function ChainModal({ onClose, open }) {
       height: "8",
       marginLeft: "8",
       width: "8"
-    })), switching && /* @__PURE__ */ React36.createElement(Box, {
+    })), switching ? networkError ? React36.createElement(Box, {
+      alignItems: "center",
+      display: "flex",
+      flexDirection: "row",
+      marginRight: "6"
+    }, /* @__PURE__ */ React36.createElement(Text, {
+      color: "error",
+      size: "14",
+      weight: "medium"
+    }, "Error switching chain")) : /* @__PURE__ */ React36.createElement(Box, {
       alignItems: "center",
       display: "flex",
       flexDirection: "row",
@@ -2398,7 +2407,7 @@ function ChainModal({ onClose, open }) {
       height: "8",
       marginLeft: "8",
       width: "8"
-    }))))), mobile && idx < (chains == null ? void 0 : chains.length) - 1 && /* @__PURE__ */ React36.createElement(Box, {
+    })): null))), mobile && idx < (chains == null ? void 0 : chains.length) - 1 && /* @__PURE__ */ React36.createElement(Box, {
       background: "generalBorderDim",
       height: "1",
       marginX: "8"
